### ISSUE TITLE
add endpoint to verify hash

### ISSUE THREAD

**Issue by wellingguzman** [directus/api#142]
> POST /utils/hash/match
> Confirms encrypted hashes against the API

**Comment by rijkvanzanten** [comment]
Is this an endpoint or a custom endpoint in the password interface?

**Comment by WellingGuzman** [comment]
An endpoint that will be located at `/[env]/utils/hash/match`

**Comment by rijkvanzanten** [comment]
Let me rephrase that to:

Why should this be a core util instead of an extension or a part of the encrypted interface?

**Comment by WellingGuzman** [comment]
What could be a useful successful response for this endpoint?

- HTTP 204 No Content
- HTTP 200 OK. Body: `{"data": true}`
- ???

Any thoughts?

**Comment by rijkvanzanten** [comment]
We could add a body like

```json
{
  "data": {
    "valid": true
  }
}
```

note that data is an object instead of a single boolean value in case we ever want to add more output to this endpoint

**Comment by WellingGuzman** [comment]
I don't think it should have data either, but some data could be useful. I will use your option.

**Comment by rijkvanzanten** [comment]
Lets make sure we use `200 OK` instead of `204 No Content` if we return that JSON 🙂
